File: post-comments/src/main/java/com/harsha/organize/PostCommentsApplication.java
package com.harsha.organize;


import com.harsha.organize.controller.DataLoadingService;
import org.springframework.boot.CommandLineRunner;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;

@SpringBootApplication
public class PostCommentsApplication implements CommandLineRunner {

    private final DataLoadingService dataLoadingService;

    public PostCommentsApplication(DataLoadingService dataLoadingService) {
        this.dataLoadingService = dataLoadingService;
    }

    public static void main(String[] args) {
        SpringApplication.run(PostCommentsApplication.class, args);

    }


    @Override
    public void run(String... args) throws Exception {
        dataLoadingService.loadPostsAndComments();
    }
}

File: post-comments 2/src/main/kotlin/com/harsha/organize/PostCommentsApplication.kt
package com.harsha.organize

import com.harsha.organize.config.WebClientConfig
import com.harsha.organize.service.DataLoadingService
import kotlinx.coroutines.runBlocking
import org.springframework.boot.CommandLineRunner
import org.springframework.boot.autoconfigure.EnableAutoConfiguration
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.runApplication
import org.springframework.context.annotation.Import
import org.springframework.data.r2dbc.repository.config.EnableR2dbcRepositories

@SpringBootApplication
@EnableR2dbcRepositories
@EnableAutoConfiguration
@Import(WebClientConfig::class)
class PostCommentsApplication(private val dataLoadingService: DataLoadingService) : CommandLineRunner {

    override fun run(vararg args: String?) {
        runBlocking {
            dataLoadingService.loadPostsAndComments()
        }
    }


}

fun main(args: Array<String>) {
    runApplication<PostCommentsApplication>(*args)
}



